kedro/framework/session/session.py

```diff
@@ -94,18 +102,20 @@ class KedroSession:
         >>>
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         session_id: str,
         package_name: str = None,
         project_path: Union[Path, str] = None,
         save_on_close: bool = False,
+        run_called: bool = False,
```

Comment:
Do we need to expose that in the constructor? What would be the usecase of someone creating a session, which is non-runnable from the start?

Reply:
What do you mean with `non-runnable`? When the pipeline is completely broken to start with?

Reply:
I think the point is that no one would ever do `KedroSession(..., run_called=True)` and so there's no need to put this argument in the constructor. Instead we could just do `self._run_called = False` on L118.

Reply:
ah okay non-runnable in that sense. Yep will change it!